pkg/state/statefile/statefile.go

```diff
@@ -245,10 +257,24 @@ func NewReader(r io.Reader, key []byte) (wire.Reader, map[string]string, error)
 		return nil, nil, err
 	}
 
-	// Wrap in compression.
-	cr, err := compressio.NewReader(r, key)
+	// Determine image compression state.
+	compressed := true
```

Comment:
Might want to add a comment here explaining why the default (if the metadata doesn't contain compression information) is the "compressed" state: Because the default behavior used to be to always compress.